R/hiper_glm.R

```diff
@@ -6,49 +6,34 @@
 #'
 #' @param design matrix, the design matrix X for the predictors.
 #' @param outcome vector, the output Y for the response.
-#' @param model the model to be used. The default is set as 'linear' to use the linear model;
-#' otherwise use 'logit' for
+#' @param model the model to be used. The default is set as 'linear' to use the linear model.
+#' @param option a list specifies the options for model solving, currently supporting 'mle_solver' only.
 #'
 #' @return A list including the information of the fitted model
 #'
 #' @export
-hiper_glm <- function(design, outcome, model = "linear") {
-  supported_model <- c("linear", "logit")
+hiper_glm <- function(design, outcome, model = "linear", option = NULL) {
+  supported_model <- c("linear")
   if (!(model %in% supported_model)) {
     stop(sprintf("The model %s is not supported.", model))
   }
-  int = F
-  if (model == 'linear'){
-    if(sum(apply(design,2, var)<=1e-10)==0){
-      design = cbind(rep(1, nrow(design)), design)
-      int = T
+  if (model == "linear") {
+    if (option$mle_solver == "pinv") {
```

Comment:
if you wanted to add a default  option, you could do add a `|is.null(option$mlesolver )` to this condition

Reply:
Similarly you may want to add error handling if a non-valid solver is supploed